gtdynamics/dynamics/Chain.cpp

```diff
@@ -85,4 +85,43 @@ Pose3 Chain::poe(const Vector &q, boost::optional<Pose3 &> fTe,
   }
   return poe;
 }
+
+Vector Chain::DynamicalEquality3(const Vector &wrench, const Vector &angles,
+                                 const Vector &torques) {
+  Matrix J;
+  poe(angles, boost::none, J);
+  return (J.transpose() * wrench - torques);
+}
+
+gtsam::Vector3_ Chain::ChainConstraint3(
+    const std::vector<JointSharedPtr> &joints, const gtsam::Key wrench_key,
+    size_t k) {
+  // Get Expression for wrench
+  gtsam::Vector6_ wrench(wrench_key);
+
+  // Get expression for joint angles as a column vector of size 3.
+  gtsam::Vector3_ angles(
+      std::bind(&Chain::MakeVector, this, std::placeholders::_1,
```

Comment:
So, (1) Expression.h has constructors for methods, so there should be no need for this binding stuff. See testExpression.h (2) MakeVector does not have a Jacobian, so where do derivatives come from?

Reply:
@dellaert  
Thinks, your comment got me thinking.
1. I changed 2 of the calls in `ChainConstraint3` to use the constructor methods from Expression.h. In the third call, I use a non-static class method (I need the members) `DynamicalEquality3` and it would not compile without std::bind.
2. So I added jacobians in MakeVector3, I think they are good. But then I thought why your question is not relevant for DynamicalEquality3 also (isn't it?). So I added jacobians there as well. I found the jacobian with respect to the joint angles quite complicated and I am still working on it.